src/fragments/lib/graphqlapi/js/working-with-files.mdx

```diff
@@ -0,0 +1,984 @@
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+async function createSong() {
+  const songDetails: CreateSongInput = {
+    name: `My first song`,
+  };
+
+  try {
+    const response = await API.graphql<GraphQLQuery<CreateSongMutation>>({
+      query: mutations.createSong,
+      variables: { input: songDetails },
+    });
+    setCurrentSong(response?.data?.createSong);
+  } catch (error) {
+    console.error("Error creating song: ", error);
+  }
+}
+``` 
+
+</Block>
+<Block name="JavaScript">
+
+```js
+async function createSong() {
+  const songDetails = {
+    name: `My first song`,
+  };
+
+  try {
+    const response = await API.graphql({
+      query: mutations.createSong,
+      variables: { input: songDetails },
+    });
+    setCurrentSong(response?.data?.createSong);
+  } catch (error) {
+    console.error("Error creating song: ", error);
+  }
+}
+```
+
+</Block>
+</BlockSwitcher>
+
+## Query the record with the GraphQL API
+
+The following helper function will be used to retrieve the record from the GraphQL API in the examples below:
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+async function getSong() {
+  if (!currentSong) return;
+
+  try {
+    const oneSong = await API.graphql<GraphQLQuery<GetSongQuery>>({
+      query: queries.getSong,
+      variables: { id: currentSong.id },
+    });
+
+    return oneSong.data?.getSong;
+  } catch (error) {
+    console.error("Error retrieving song: ", error);
+  }
+}
+``` 
+
+</Block>
+<Block name="JavaScript">
+
+```js
+async function getSong() {
+  if (!currentSong) return;
+
+  try {
+    const oneSong = await API.graphql({
+      query: queries.getSong,
+      variables: { id: currentSong.id },
+    });
+
+    return oneSong.data?.getSong;
+  } catch (error) {
+    console.error("Error retrieving song: ", error);
+  }
+}
+```
+
+</Block>
+</BlockSwitcher>
+
+## Uploading the file to S3
+
+Create a helper function that will handle uploading the file to S3:
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+async function uploadImage(e: React.ChangeEvent<HTMLInputElement>) {
+  if (!e.target.files) return;
+
+  const file = e.target.files[0];
+
+  try {
+    const result = await Storage.put(file.name, file, {
+      contentType: "image/png", // contentType is optional
+    });
+
+    return result?.key;
+  } catch (error) {
+    console.error("Error uploading image: ", error);
+  }
+}
+``` 
+
+</Block>
+<Block name="JavaScript">
+
+```js
+async function uploadImage(e) {
+  if (!e.target.files) return;
+
+  const file = e.target.files[0];
+
+  try {
+    const result = await Storage.put(file.name, file, {
+      contentType: "image/png", // contentType is optional
+    });
+
+    return result?.key;
+  } catch (error) {
+    console.error("Error uploading image: ", error);
+  }
+}
+```
+
+</Block>
+</BlockSwitcher>
+
+## Adding / updating an associated file
```

Comment:
In the `meta.description`, the word "attachment" is used which is inconsistent with "associated". Let's just pick one and remain consistent. I prefer attached/attachment but not strongly opinionated.

Reply:
The reason I chose "file" / "association" over "attachment" / "attached" was because I didn't want to confuse customers regarding what was actually happening here. We are creating an association between the record and the file, which can be later updated, but there is otherwise no library logic that connects these two in any way (unlike the AppSync SDK, which was performing the upload), and I was concerned that "attachment" may imply that. My preference would be to stick with "file / association", if that works for you.

Reply:
Cool. Let's then use association throughout.